src/TestProblem/Hydro/DynamicalFriction/Init_TestProb_Hydro_DynamicalFriction.cpp

```diff
@@ -0,0 +1,448 @@
+   } // for (int lv=0; lv<NLEVEL; lv++)
+}
+
+//-------------------------------------------------------------------------------------------------------
+// Function    :  Aux_Record_User_DynamicalFriction
+// Description :  Update the GC position and Halo position in each step
+//
+// Note        :  1. Invoked by main() using the function pointer "Aux_Record_User_Ptr",
+//                   which must be set by a test problem initializer
+//                2. Enabled by the runtime option "OPT__RECORD_USER"
+//                3. This function will be called both during the program initialization and after each full update
+//                4. Notice that this function can only get "ONE" GC, multiple GCs are not supported.
+//                5. Write to the file "Record__Result_GC_position" and "Record__Result_Center"
+// Parameter   :  None
+//-------------------------------------------------------------------------------------------------------
+void Aux_Record_User_DynamicalFriction()
+{
+
+// A. Find the GC's position
+   double GC_x_local = 0.0, GC_y_local = 0.0, GC_z_local = 0.0;
+   double GC_x, GC_y, GC_z;
+
+
+
+   for (long p=0; p<amr->Par->NPar_AcPlusInac; p++) {
+   if ( amr->Par->Mass[p] < (real)0.0 )  continue;
+      if ( amr->Par->Type[p] == PTYPE_GC)
+      {
+         GC_x_local = amr->Par->PosX[p];
+         GC_y_local = amr->Par->PosY[p];
+         GC_z_local = amr->Par->PosZ[p];
+
+         char Filename[MAX_STRING];
+         sprintf( Filename, "%s", "Record__Result_GC_position" );
+         FILE *File = fopen( Filename, "a" );
+         if ( Time[0]==0.0 )
+         {
+           fprintf(File, "%15s\t%15s\t%15s\t%15s\n", "#          Time", " GCX", " GCY", " GCZ");
+         }
+         fprintf(File, "%15.7f\t%15.7e\t%15.7e\t%15.7e\n", Time[0],GC_x_local,GC_y_local,GC_z_local );
+         fclose ( File );
+      }
+   }
+
+// B. Broadcast the GC's position to all rank
+   MPI_Allreduce(&GC_x_local, &GC_x, 1, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD);
+   MPI_Allreduce(&GC_y_local, &GC_y, 1, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD);
+   MPI_Allreduce(&GC_z_local, &GC_z, 1, MPI_DOUBLE, MPI_MAX, MPI_COMM_WORLD);
+   
+   MPI_Bcast(&GC_x,1,MPI_DOUBLE,0,MPI_COMM_WORLD);
+   MPI_Bcast(&GC_y,1,MPI_DOUBLE,0,MPI_COMM_WORLD);
+   MPI_Bcast(&GC_z,1,MPI_DOUBLE,0,MPI_COMM_WORLD);   
+   MPI_Barrier(MPI_COMM_WORLD);
+
+
+// C. Set the center finding method base on Input__TestProb
+   if ( FixCenter )
+   {
+      if ( MPI_Rank == 0 )
+         {
+         char Filename[MAX_STRING];
+         sprintf( Filename, "%s", "Record__Result_Center" );
+         FILE *File = fopen( Filename, "a" );
+         if ( Time[0]==0.0 )
+         {
+            fprintf(File, "%15s\t%15s\t%15s\t%15s\n", "#          Time", " CenterX", " CenterY", " CenterZ");
+         }
+         fprintf(File, "%15.7f\t%15.7e\t%15.7e\t%15.7e\n",Time[0]  ,amr->BoxCenter[0], amr->BoxCenter[1], amr->BoxCenter[2]);
+         fclose ( File );
+      }
+   }
```

Comment:
When `FixCenter` is on, it seems it just keeps recording the box centers, which are all constants.
It seems unnecessary and wasting time to me.
How about *always* recording the minimum potential in `Record__Result_Center`?
When we need to fix the center in the post-analysis, we can easily ignore `Record__Result_Center` and use the box center as the center instead.
Then, I think we don't need the parameter `FixCenter` in the simulations.